sqlglot/optimizer/scope.py

```diff
@@ -621,19 +621,23 @@ def _is_derived_table(expression: exp.Subquery) -> bool:
 def _traverse_tables(scope):
     sources = {}
 
+    scope_expression = scope.expression
+    if isinstance(scope_expression, exp.Subquery):
+        scope_expression = scope_expression.unnest()
```

Comment:
is unnesting always safe? what if we have joins or order bys etc? or will those never happen?

Reply:
That's a good point, I'll need to investigate a bit more. It seems like `scope.tables` has a similar problem so will have to figure that out too.